### DISCUSSION TITLE
proposal: custom make/patch/kconfig for static addons

### DISCUSSION THREAD

**Discussion by mgliewe** [Freetz-NG/freetz-ng#499] [Allgemein]
Hi,
fuer unseren Verein bastle ich gerade an einem captive portal, das auf der fritzOS Kindersicherung basiert. Ich benutze die luacgi der original Weboberflaeche. Das ist notwendig, da ich Zugriff auf die AVM Firewall und die Zugangsprofile brauche (via box.query/box.set_config) und auch die originale Kindersicherungsseite abfangen muss. iptables geht ja leider nicht, waere mir lieber gewesen.

Klappt auch alles super, der buildprozess ist allerdings etwas umstaendlich. 

Momentan spiele ich das als statisches Addon in die Firmware ein. Ich benoetige allerdings einige native lua-module, und einige patches der original-lsp Dateien.

D.h.: git update, make fuer freetz, dann den build des addons anwerfen, anschliessend noch einmal freetz build, dann flashen. 

Gibt es eine geschicktere Moeglichkeit? Waere es sinnvoll custom Makefiles/patches fuer statische Pakete in fwmod einzubauen?
und wenn mensch schon dabei ist, ev sogar eigene kconfigs?

Das Freetz Buildsystem waere zwar an sich super fuer den Anwendungsfall geeignet, ist aber doch etwas unhandlich, und ich wuerde den git sourcetree mit meiner proprietaeren Anwendung verschmutzen, was updates nicht einfacher macht....

lg maG

**Comment by fda77** [comment]
Ich verstehe die Frage nicht so ganz. Weshalb baust du nicht gleich alles ein?

**Comment by mgliewe** [comment]
naja, ich brauche zum einem proprietare lua module (zB crypto fuers authentifizieren, sha2/md5 in pure lua5.1 ist pain, lfs und sqlite benutze ich auch grad).. 

die kann ich statisch lassen, da wird sich nicht viel tun.

ich muss aber auch die index.lua und kids_not_allowed.lua der original firmware patchen um auf captive portal umzuleiten. 
Wenn sich von Seiten avm da was aendert (wie gerade beim update auf 7.29 massiv geschehen) und ich in nem halbem jahr da nochmal - kannste mal kurz - hand anlege... da sind statische dateien nicht so guenstig.

kann ich mit entprechenden patch-files in einem privatem branch loesen. Dachte mir nur, ich waere nicht der einzige der auf solch ein Problem stoesst.
